src/main/java/aegis/Aegis.java

```diff
@@ -0,0 +1,282 @@
+
+        System.out.println(SEPARATOR);
+        System.out.println("Hello! This is Aegis, an Anti-Shadow Suppression Weapon and a chatbot.");
+        System.out.println("What can I do for you?");
+
+        ArrayList<Task> taskList = new ArrayList<>();
+        loadTasksFromFile(taskList);
+
+        while (true) {
+            System.out.println();
+            System.out.println("---------------BEGIN--OF--PROMPT------------------");
+
+            String input = scanner.nextLine().trim();
+
+            if (input.equalsIgnoreCase("bye")) {
+                break;
+            }
+
+            handleCommand(input, taskList);
+        }
+
+        System.out.println(" Bye. Hope to see you again soon!");
+        System.out.println();
+        System.out.println("-------------END--OF--CONVERSATION----------------");
+
+        scanner.close();
+    }
+
+    private static void handleCommand(String input, ArrayList<Task> taskList) {
+        try {
+            String command = input.split(" ")[0].toLowerCase();
+            switch (command) {
+            case "list":
+                displayTaskList(taskList);
+                break;
+            case "mark":
+                markTaskAsDone(taskList, input);
+                break;
+            case "unmark":
+                unmarkTaskAsDone(taskList, input);
+                break;
+            case "deadline":
+                addDeadline(taskList, input);
+                break;
+            case "event":
+                addEvent(taskList, input);
+                break;
+            case "todo":
+                addTodoTask(taskList, input);
+                break;
+            case "delete":
+                deleteTask(taskList, input);
+                break;
+            default:
+                throw new AegisException(" Your command has not been authorized ");
+            }
+        } catch (AegisException e) {
+            System.out.printf(" Anomaly detected: %s%n", e.getMessage());
+        }
+    }
+
+    private static void displayTaskList(ArrayList<Task> taskList) {
+        System.out.println(" I could do much more than checking your list, but, if you wish:");
+        for (int i = 0; i < taskList.size(); i++) {
+            System.out.printf("   %d. %s%n", i + 1, taskList.get(i));
+        }
+    }
+
+    private static void markTaskAsDone(ArrayList<Task> taskList, String input) {
+        try {
+            int taskIndex = Integer.parseInt(input.split(" ")[1]) - 1;
+            if (taskIndex < 0 || taskIndex >= taskList.size()) {
+                throw new AegisException("Invalid task number");
+            }
+            taskList.get(taskIndex).markAsDone();
+            saveTasksToFile(taskList);
+            System.out.printf(" I've marked this task as done:%n   %s%n", taskList.get(taskIndex));
+        } catch (AegisException | NumberFormatException e) {
+            System.out.printf(" Anomaly detected: [%s]%n", e.getMessage());
+        }
```

Comment:
You may consider abstracting the content into separate methods and pass in parameters to differentiate as the code pattern is similar for ```unmarkTaskAsDone``` method.